packages/calypso-e2e/src/lib/pages/login-page.ts

```diff
@@ -156,18 +160,43 @@ export class LoginPage {
 		credentials: LoginCredentials | TestAccount,
 		options?: LoginOptions
 	): Promise< void > {
-		await this.visit();
-		await this.revealLoginForm();
-
-		// Obtain sanitized username/password combination.
-		const { username, password } = await this.resolveUserCredentials( credentials );
-
 		// The `waitForNavigation` method triggered after clicking on the Log In button will wait for a
 		// customn URL if `options.landingUrl` is specified.
 		// This is useful if after logging into Calypso the redirect takes user away from the default
 		// redirect of `<host>/home/<blogUrl>.
 		const landingUrl = options?.landingUrl ? options.landingUrl : `**/home/**`;
 
+		// If there is a stored cookie for the user, try that first.
+		if ( 'account' in credentials ) {
```

Comment:
One thought I had was to split the `login` methods to two overloaded methods - one accepting `LoginCredentials` type and the other accepting a `TestAccount` type.

It's been ages since I had to use overloading so I went to look it up, and it seems TypeScript doesn't have a super intuitive overloading (a là Java 7).

As such I've restructured this method to perform both 1) cookie authentication attempt and 2) normal authentication.

Reply:
>it seems TypeScript doesn't have a super intuitive overloading (a là Java 7)
Yeah, JS/TS overloading is basically just doing what you're doing here - you have to have one function that handles all the argument options!

I think this function looks great - it's very linear and clear!! 🎉